lib/web/apiserver.go

```diff
@@ -806,8 +806,9 @@ func (h *Handler) bindDefaultEndpoints() {
 	// Fetches the user's preferences
 	h.GET("/webapi/user/preferences", h.WithAuth(h.getUserPreferences))
 
-	// Updates the user's preferences
-	h.PUT("/webapi/user/preferences", h.WithAuth(h.updateUserPreferences))
+	// Updates the user's preferences. This leverages WithAuthCookieAndCSRF rather than WithAuth because
+	// we update preferences during the onboarding flow, before a bearer token is set.
+	h.PUT("/webapi/user/preferences", h.WithAuthCookieAndCSRF(h.updateUserPreferences))
```

Comment:
I would love thoughts on this change. Are we still secure if leveraging `WithAuthCookieAndCSRF`?

We set a bearer token on redirect to `/web` at the end of the registration. Until then, there is no bearer token to check.

Reply:
tbh, i'm not sure either... i would love to know a definite answer though, because i have questions, like why not all protected endpoints use that too then? (perhaps these endpoints where it's used, the consequence aren't as high?) and also why didn't we pass the access token instead of the csrf token in the form? (this form is located where user was already authenticated)

but i took a look at `WithAuthCookieAndCSRF` and it's intended for `non-ajax` request (forms) so I would generally stay away from it. 

i wonder if we can use local storage to temporarily store the user preference, then on [init](https://github.com/gravitational/teleport/blob/31b66f092252ee07fc7a41ca0d4924a1f91c1b3c/web/packages/teleport/src/teleportContext.tsx#L77), we can update user preference there, and then delete that storage key?